modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -87,15 +97,24 @@ def report msg, response, error_only
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
 
-    def match_zone(fqdn)
+    def match_zone(fqdn_or_ip)
```

Comment:
You just made this method responsible for determining the zone name. This logic is implicit in the context of the call to ```match_zone```: create_a/remove_a methods use fqdn, create_ptr/remove_ptr need a ptr. I'd also pass in zone list as a parameter while I'm at it -- makes testing easier. So, for the former:

```
match_zone(fqdn, enum_zones)
```

for ptr-related methods:
```
match_zone(get_ptr_record(fqdn_or_ip), enum_zones)
```